packages/client/src/schema/repository.ts

```diff
@@ -348,7 +348,7 @@ export class RestRepository<Data extends BaseData, Record extends XataRecord = D
 {
   #table: string;
   #getFetchProps: () => Promise<FetcherExtraProps>;
-  db: SchemaPluginResult<any>;
+  #db: SchemaPluginResult<any>;
```

Comment:
🤔 Is this change required for the new `getConfig()` method or is this something you wanted to do anyway?

Should we maek this readonly somehow?

Reply:
Not related, but I noticed it while doing this change.

We don't have the db object in the interface definition. So was available because of JS even if not typed in TS. We shouldn't expose the internals and only public facing API.